[new-parser] CEP rules: broken annotations

## Parent issue
- #5678 

## Failing tests
- `org.drools.model.codegen.execmodel.CepTest#testExpires`

## Notes
- Both `STANDARD_FROM_DRL` and `PATTERN_DSL`.

## Rule code snippet
```
declare StockFact
    @role( value = event )
    @expires( value = 2s, policy = TIME_SOFT )
end
```

## Error output
```
### parse : ANTLR4_PARSER_ENABLED = true
line 6:22 no viable alternative at input 'value = 2s'
line 6:20 mismatched input '=' expecting ':'
line 6:45 extraneous input ')' expecting 'end'
14:50:01.178 [main] ERROR o.d.c.k.b.impl.AbstractKieProject.buildKnowledgePackages:280 - Unable to build KieBaseModel:kb
[6,22]: no viable alternative at input 'value = 2s'
[6,20]: mismatched input '=' expecting ':'
[6,45]: extraneous input ')' expecting 'end'
[0,0]: Parser returned a null Package


java.lang.AssertionError: [Message [id=1, kieBase=kb, level=ERROR, path=r0.drl, line=6, column=0
   text=no viable alternative at input 'value = 2s'], Message [id=2, kieBase=kb, level=ERROR, path=r0.drl, line=6, column=0
   text=mismatched input '=' expecting ':'], Message [id=3, kieBase=kb, level=ERROR, path=r0.drl, line=6, column=0
   text=extraneous input ')' expecting 'end'], Message [id=4, kieBase=kb, level=ERROR, path=r0.drl, line=0, column=0
   text=Parser returned a null Package]]

	at org.drools.model.codegen.execmodel.BaseModelTest.createKieBuilder(BaseModelTest.java:157)
	at org.drools.model.codegen.execmodel.BaseModelTest.createKieBuilder(BaseModelTest.java:132)
	at org.drools.model.codegen.execmodel.BaseModelTest.getKieContainer(BaseModelTest.java:121)
	at org.drools.model.codegen.execmodel.BaseModelTest.getKieContainer(BaseModelTest.java:114)
	at org.drools.model.codegen.execmodel.BaseModelTest.getKieSession(BaseModelTest.java:110)
	at org.drools.model.codegen.execmodel.CepTest.testExpires(CepTest.java:452)
```

--- a/drools-drl/drools-drl-parser-tests/src/test/java/org/drools/drl/parser/antlr4/MiscDRLParserTest.java
+++ b/drools-drl/drools-drl-parser-tests/src/test/java/org/drools/drl/parser/antlr4/MiscDRLParserTest.java
@@ -3749,9 +3749,12 @@ class MiscDRLParserTest {
                 "then\n" +
                 "end\n";
         PackageDescr packageDescr = parser.parse(text);
+        assertThat(parser.hasErrors()).as(parser.getErrors().toString()).isFalse();
+
         RuleDescr ruleDescr = packageDescr.getRules().get(0);
         PatternDescr patternDescr = (PatternDescr) ruleDescr.getLhs().getDescrs().get(0);
         AnnotationDescr annotationDescr = patternDescr.getAnnotations().iterator().next();
+
         assertThat(annotationDescr.getName()).isEqualTo("watch");
         assertThat(annotationDescr.getSingleValueAsString()).isEqualTo("!*, age");
     }
@@ -3769,5 +3772,25 @@ class MiscDRLParserTest {
         PatternDescr patternDescr = (PatternDescr) ruleDescr.getLhs().getDescrs().get(0);
         FromDescr fromDescr = (FromDescr) patternDescr.getSource();
         assertThat(fromDescr.getDataSource().toString()).isEqualTo("new Person(\"John\", 30)");
+    }
+
+    @Test
+    public void expiresWithTimeLiteralValue() {
+        String text = "package org.drools\n" +
+                "declare StockFact\n" +
+                "    @role( value = event )\n" +
+                "    @expires( value = 2s, policy = TIME_SOFT )\n" +
+                "end";
+
+        PackageDescr packageDescr = parser.parse(text);
+        assertThat(parser.hasErrors()).as(parser.getErrors().toString()).isFalse();
+
+        AnnotationDescr annotationDescr = packageDescr
+                .getTypeDeclarations().get(0)
+                .getAnnotation("expires");
+
+        assertThat(annotationDescr.getSingleValueAsString()).isEqualTo("2s");
+        assertThat(annotationDescr.getValueAsString("value")).isEqualTo("2s");
+        assertThat(annotationDescr.getValueAsString("policy")).isEqualTo("TIME_SOFT");
     }
 }

--- a/drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4
+++ b/drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4
@@ -321,6 +321,7 @@ drlLiteral
     | BOOL_LITERAL
     | NULL_LITERAL
     | TEXT_BLOCK // Java17
+    | TIME_INTERVAL
     ;
 
 inlineListExpression

